lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -134,9 +125,41 @@ public void dropData() throws IOException {
             }
         }
     }
-    
-    /* checks that at least one index exists for a type, and throws an exception otherwise.
-     * Returns a index string that matches any version of the type.
+
+
+    /** The specified list is valid if it,
+     *
+     * 1. is empty,
```

Comment:
I'm fine with the ElasticIndexingStorage level rejecting null search lists, but I do think that leaving the object_types field out of the incoming json should result in a non-type-constrained search rather than an illegal argument exception. If that behavior is controlled in a layer above EIS that's ok with me.

Reply:
I have added a bit of validation to ensure that an empty list is created if it is left out in the request from the client.